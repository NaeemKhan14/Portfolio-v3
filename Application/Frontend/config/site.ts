export type SiteConfig = typeof siteConfig

export const siteConfig = {
  name: 'Naeem Khan',
  description: "Naeem Ullah Khan's personal portfolio",
  navItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'Certificates',
      href: '/certificates',
    },
    {
      label: 'Projects',
      href: '/projects',
    },
    {
      label: 'Blog',
      href: '/blog',
    },
  ],
  navMenuItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'Certificates',
      href: '/certificates',
    },
    {
      label: 'Projects',
      href: '/projects',
    },
    {
      label: 'Blog',
      href: '/blog',
    },
  ],
  links: {
    linkedin: 'https://www.linkedin.com/in/naeem-khan-/',
    github: 'https://github.com/NaeemKhan14/',
  },
}
