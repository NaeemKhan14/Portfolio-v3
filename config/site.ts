export type SiteConfig = typeof siteConfig;

export const siteConfig = {
  name: "Naeem Khan",
  description: "Naeem Ullah Khan's personal portfolio",
  navItems: [
    {
      label: "Home",
      href: "/",
    },
    {
      label: "Projects",
      href: "/projects",
    },
    {
      label: "Blog",
      href: "/blog",
    },
  ],
  navMenuItems: [
    {
      label: "Home",
      href: "/",
    },
    {
      label: "Blog",
      href: "/blog",
    },
    {
      label: "Projects",
      href: "/projects",
    },

  ],
  links: {
    github: "https://github.com/NaeemKhan14/",
    twitter: "https://twitter.com/hero_ui",
    docs: "https://heroui.com",
    discord: "https://discord.gg/9b6yyZKmH4",
    sponsor: "https://patreon.com/jrgarciadev",
  },
};
