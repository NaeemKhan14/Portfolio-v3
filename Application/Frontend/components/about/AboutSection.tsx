'use client'

import { siteConfig } from '@/config/site'
import Image from 'next/image'
import { GithubIcon, LinkedinIcon } from '../ui/icons'
import Link from 'next/link'
import { Chip, Divider } from '@heroui/react'
import DownloadButton from '@/hooks/DownloadButton'

export default function AboutSection() {
  return (
    <section className='py-8'>
      <div className='mx-auto mb-10 max-w-3xl px-6'>
        <div className='grid grid-cols-1 items-center gap-8 md:grid-cols-[1fr_2fr]'>
          {/* Profile Section */}
          <div className='flex flex-col items-center space-y-6'>
            <div className='relative'>
              <Image
                src='/images/pp.jpg'
                alt='Profile photo'
                className='border-warning-300 dark:border-warning/50 rounded-2xl border-2 shadow-sm'
                width={200}
                height={200}
                priority
                unoptimized
              />
              <div className='text-center'>
                <Chip
                  color='warning'
                  variant='solid'
                  className='dark:text-danger-300 absolute -bottom-3 left-1/2 -translate-x-1/2 text-sm font-medium text-pink-700'
                >
                  SOC Analyst & Software Dev
                </Chip>
              </div>
            </div>

            <div className='flex w-full flex-col items-center space-y-4'>
              <div className='flex space-x-4'>
                <Link
                  target='_blank'
                  aria-label='Linkedin'
                  href={siteConfig.links.linkedin}
                  className='rounded-full bg-gray-100 p-3 transition-transform hover:scale-125 hover:bg-gray-200 dark:bg-gray-800 dark:hover:bg-gray-700'
                >
                  <LinkedinIcon className='h-5 w-5 text-gray-700 dark:text-gray-300' />
                </Link>
                <Link
                  target='_blank'
                  aria-label='Github'
                  href={siteConfig.links.github}
                  className='rounded-full bg-gray-100 p-3 transition-transform hover:scale-125 hover:bg-gray-200 dark:bg-gray-800 dark:hover:bg-gray-700'
                >
                  <GithubIcon className='h-5 w-5 text-gray-700 dark:text-gray-300' />
                </Link>
              </div>

              <div className='grid w-full grid-cols-2 gap-3 md:grid-cols-1'>
                <DownloadButton
                  label='Cybersecurity CV'
                  filename='Naeem Khan CV (Cyber Security).pdf'
                />
                <DownloadButton
                  label='Software Developer CV'
                  filename='Naeem Khan CV.pdf'
                />
              </div>
            </div>
          </div>

          {/* Content Section */}
          <div className='text-center lg:text-left'>
            <h1 className='text-danger mb-4 text-3xl font-bold'>Naeem Khan</h1>
            <p className='mb-6 leading-relaxed text-gray-700 dark:text-gray-300'>
              Cybersecurity specialist with software development expertise,
              focused on building secure systems and defending against modern
              threats. Combining SOC analyst certification with engineering
              experience to create resilient applications.
            </p>
            <div
              role='list'
              className='flex flex-wrap justify-center gap-2 lg:justify-start'
            >
              <Chip role='listitem' color='danger' variant='bordered'>
                Software Development
              </Chip>
              <Chip role='listitem' color='danger' variant='bordered'>
                SOC Analyst L1
              </Chip>
              <Chip role='listitem' color='danger' variant='bordered'>
                CTF Competitor
              </Chip>
            </div>
          </div>
        </div>
      </div>
      <Divider className='mx-auto max-w-3xl' />
    </section>
  )
}
