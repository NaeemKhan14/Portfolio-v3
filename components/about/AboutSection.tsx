import { siteConfig } from "@/config/site";
import Image from "next/image";
import { GithubIcon, LinkedinIcon } from "../icons";
import { Link } from "@heroui/link";

export default function AboutSection() {
  return (
    <section className="flex flex-col items-center justify-center py-8 md:py-10">
      <div className="flex flex-col-reverse lg:flex-row max-w-3xl w-full px-8 gap-8">
        
        {/* Text Block */}
        <div className="prose prose-neutral text-xl dark:prose-invert flex-1">
          <h1 className="mb-8 text-2xl font-medium text-center">About Me</h1>
          <p>
            Hello, I'm Naeem Khan, a cybersecurity-focused software developer based in Berlin. After earning my Master's in Computer Science with Cybersecurity specialization from SRH Berlin University, I actively pursue industry certifications (SAL1, PT1) and participate in Capture The Flag competitions to sharpen my real-world vulnerability assessment skills against evolving threats.
          </p>
          <br />
          <p>
            My cybersecurity expertise is reinforced by two years of software development experience in Dubai, where I engineered full-stack solutions with Python, Django, PostgreSQL, and Angular. This dual perspective enables me to build inherently secure systems from the ground up; transforming security theory into resilient applications that withstand modern cyber risks.
          </p>
        </div>

        {/* Avatar & Links/Buttons */}
        <div className="flex flex-col items-center justify-start gap-4 mt-4 sm:mt-14 ">
          {/* Row 1: Avatar */}
          <Image
            src="images/pp.jpg"
            alt="Profile photo"
            className="rounded-full"
            unoptimized
            width={210}
            height={210}
            priority
          />

          {/* Row 2: Social Icons */}
          <div className="flex gap-4">
            <Link isExternal aria-label="Linkedin" href={siteConfig.links.linkedin}>
              <LinkedinIcon className="w-6 h-6 text-red-600 dark:text-red-400 hover:text-pink-300 dark:hover:text-pink-900" />
            </Link>
            <Link isExternal aria-label="Github" href={siteConfig.links.github}>
              <GithubIcon className="w-6 h-6 text-red-600 dark:text-red-400 hover:text-pink-300 dark:hover:text-pink-900" />
            </Link>
          </div>

          {/* Row 3: Buttons */}
          <div className="flex gap-3">
            <a href="cv/Naeem Khan CV (Cyber Security).pdf" className="text-red-600 dark:text-red-400 bg-red-50 dark:bg-red-900/30 px-8 py-2 rounded-lg text-sm hover:bg-pink-300 dark:hover:bg-pink-900">
              CV 1
            </a>
            <a href="cv/Naeem Khan CV.pdf" className="text-red-600 dark:text-red-400 bg-red-50 dark:bg-red-900/30 px-8 py-2 rounded-lg text-sm hover:bg-pink-300 dark:hover:bg-pink-900">
              CV 2
            </a>
          </div>
        </div>
      </div>
    </section>
  );
}
