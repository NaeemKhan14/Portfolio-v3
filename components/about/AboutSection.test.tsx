import { render, screen } from "@testing-library/react";
import AboutSection from "./AboutSection";
import { siteConfig } from "@/config/site";
import '@testing-library/jest-dom'

describe("AboutSection", () => {
    beforeEach(() => {
        render(<AboutSection />);
    });

    test("renders section heading", () => {
        expect(screen.getByRole("heading", { name: /about me/i })).toBeInTheDocument();
    });

    test("renders profile image", () => {
        const image = screen.getByAltText("Profile photo");
        expect(image).toBeInTheDocument();
        expect(image).toHaveAttribute("src", expect.stringContaining("/images/pp.jpg"));
    });

    test("renders description text", () => {
        expect(
            screen.getByText(/I'm Naeem Khan, a cybersecurity-focused software developer/i)
        ).toBeInTheDocument();
        expect(
            screen.getByText(/This dual perspective enables me to build inherently secure systems/i)
        ).toBeInTheDocument();
    });

    test("renders social links", () => {
        const linkedin = screen.getByLabelText("Linkedin");
        const github = screen.getByLabelText("Github");

        expect(linkedin).toBeInTheDocument();
        expect(linkedin).toHaveAttribute("href", siteConfig.links.linkedin);

        expect(github).toBeInTheDocument();
        expect(github).toHaveAttribute("href", siteConfig.links.github);
    });

    test("renders CV download buttons with correct hrefs", () => {
        const cv1 = screen.getByRole("link", { name: /cv 1/i });
        const cv2 = screen.getByRole("link", { name: /cv 2/i });

        expect(cv1).toBeInTheDocument();
        expect(cv1).toHaveAttribute("href", "cv/Naeem Khan CV (Cyber Security).pdf");

        expect(cv2).toBeInTheDocument();
        expect(cv2).toHaveAttribute("href", "cv/Naeem Khan CV.pdf");
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<AboutSection />);
        expect(asFragment()).toMatchSnapshot();
    });
});
