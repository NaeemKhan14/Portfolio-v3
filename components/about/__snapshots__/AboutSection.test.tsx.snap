// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AboutSection matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="flex flex-col items-center justify-center py-8 md:py-10"
  >
    <div
      class="flex flex-col-reverse lg:flex-row max-w-3xl w-full px-8 gap-8"
    >
      <div
        class="prose prose-neutral text-xl dark:prose-invert flex-1"
      >
        <h1
          class="mb-8 text-2xl font-medium text-center"
        >
          About Me
        </h1>
        <p>
          Hello, I'm Naeem Khan, a cybersecurity-focused software developer based in Berlin. After earning my Master's in Computer Science with Cybersecurity specialization from SRH Berlin University, I actively pursue industry certifications (SAL1, PT1) and participate in Capture The Flag competitions to sharpen my real-world vulnerability assessment skills against evolving threats.
        </p>
        <br />
        <p>
          My cybersecurity expertise is reinforced by two years of software development experience in Dubai, where I engineered full-stack solutions with Python, Django, PostgreSQL, and Angular. This dual perspective enables me to build inherently secure systems from the ground up; transforming security theory into resilient applications that withstand modern cyber risks.
        </p>
      </div>
      <div
        class="flex flex-col items-center justify-start gap-4 mt-4 sm:mt-14 "
      >
        <img
          alt="Profile photo"
          class="rounded-full"
          data-nimg="1"
          decoding="async"
          height="210"
          src="/images/pp.jpg"
          style="color: transparent;"
          width="210"
        />
        <div
          class="flex gap-4"
        >
          <a
            aria-label="Linkedin"
            class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-hover active:opacity-disabled transition-opacity"
            data-react-aria-pressable="true"
            href="https://www.linkedin.com/in/naeem-khan-/"
            rel="noopener noreferrer"
            role="link"
            tabindex="0"
            target="_blank"
          >
            <svg
              class="w-6 h-6 text-danger hover:text-pink-300 dark:hover:text-pink-900"
              height="24"
              viewBox="0 0 448 512"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            aria-label="Github"
            class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-hover active:opacity-disabled transition-opacity"
            data-react-aria-pressable="true"
            href="https://github.com/NaeemKhan14/"
            rel="noopener noreferrer"
            role="link"
            tabindex="0"
            target="_blank"
          >
            <svg
              class="w-6 h-6 text-danger hover:text-pink-300 dark:hover:text-pink-900"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
        <div
          class="flex gap-3"
        >
          <a
            class="text-white bg-danger px-8 py-2 rounded-lg text-sm hover:bg-pink-300 dark:hover:bg-pink-900"
            href="cv/Naeem Khan CV (Cyber Security).pdf"
          >
            CV 1
          </a>
          <a
            class="text-white bg-danger px-8 py-2 rounded-lg text-sm hover:bg-pink-300 dark:hover:bg-pink-900"
            href="cv/Naeem Khan CV.pdf"
          >
            CV 2
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
